gsr_booking/booking_logic.py

```diff
@@ -0,0 +1,255 @@
+import requests
+import datetime
+import math
+import random
+
+
+class BookingLogic:
+
+    DATE_FORMAT_STR = "%Y-%m-%dT%H:%M:%S%z"
+    MAX_SLOT_HRS = 2.0  # the longest booking allowed per person
+    MIN_SLOT_HRS = 0.5  # the minimum booking allowed per person
+
+    def book_room_for_group(self, group, is_wharton, room, lid, start, end, requester_pennkey):
+        # makes a request to labs api server to book rooms, and returns result json if successful
+        members = group.get_pennkey_active_members()
+        if len(members) < 1:
+            return {
+                "complete_success": False,
+                "partial_success": False,
+                "error": "No members in the group have enabled their pennkey to be used for group bookings",
+            }
+
+        # randomize order, and put requester first in the order
```

Comment:
Don't know if you want this to be a TODO to work on a more nuanced booking algorithm

Reply:
I know we have talked about a more nuanced booking algorithm before, but I'm curious what that entails specifically?